run

```diff
@@ -2,6 +2,18 @@
 set -x
 set -eo pipefail
 
+###
+# This is specific solution for wildfly problems
+# Ref: https://github.com/redhat-developer/odo/issues/445
+if [ -e ${ODO_S2I_DEPLOYMENT_DIR}/ROOT.war ]; then
```

Comment:
you can't have `ROOT.war` hardcoded, the file can have any name, it even doesn't have to be `.war`.

We are not trying to hack this just for wildfly but create generic solution